### PR TITLE
feat: select profile/entity in url

### PR DESCRIPTION
**Pull request by Rom1-B**
Force the selection of an entity and/or a profile in the URL

parameters:
- change_profile
- change_entity

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !21545

### CODE REVIEW

**Review by orthagh — APPROVED** [review]

**Review by trasher — APPROVED** [review]

File: src/CommonGLPI.php [hunk 1]
```diff
@@ -1057,6 +1057,28 @@ class='btn btn-sm btn-icon btn-ghost-secondary'
    function display($options = []) {
       global $CFG_GLPI;
 
+      $redirect = false;
+
+      // Manage profile change
+      if (isset($_GET["change_profile"])) {
```
Comment (cedric-anne) [inline src/CommonGLPI.php:1063]: No need to redirect if profile is same as the active one.
```suggestion
      if (isset($_GET["change_profile"]) && ($_SESSION['glpiactiveprofile']['id'] ?? -1) != $_GET["change_profile"]) {
```

File: src/CommonGLPI.php [hunk 2]
```diff
@@ -1057,6 +1057,28 @@ class='btn btn-sm btn-icon btn-ghost-secondary'
...
+         if (isset($_SESSION['glpiprofiles'][$_GET["change_profile"]])) {
+            Session::changeProfile($_GET["change_profile"]);
+         }
+         $redirect = true;
+      }
+
+      // Manage entity change
+      if (isset($_GET["change_entity"])) {
```
Comment (cedric-anne) [inline src/CommonGLPI.php:1071]:
```suggestion
      if (isset($_GET["change_entity"]) && ($_SESSION["glpiactive_entity"] ?? -1) != $_GET["change_entity"]) {
```

File: src/CommonGLPI.php [hunk 3]
```diff
@@ -1057,6 +1057,28 @@ class='btn btn-sm btn-icon btn-ghost-secondary'
...
+         Session::changeActiveEntities($_GET["change_entity"], true);
+         $redirect = true;
+      }
+
+      if ($redirect) {
+         Html::redirect(
+            preg_replace("/(\?|&|".urlencode('?')."|".urlencode('&').")?(change_entity|change_profile).*/",
```
Comment (cedric-anne) [inline src/CommonGLPI.php:1078]: 1. If `change_entity` or `change_profile` is placed before some orther parameters, they will be removed.
2. I guess you used `HTTP_REFERER` to users that are not already logged-in, but for users that do not have to use login form, should this logic be applied to `REQUEST_URI` and `QUERY_STRING` keys ?

**Review by cedric-anne — COMMENTED** [review]
Not really sure that `CommonGLPI::display()` is the best place for implementing this.

If it is used to append a custom parameter to notification URLs, then it should probably be handled like the `redirect` param in the `/index.php` script.

**Review by trasher — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Pull request merged**